src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,24 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.readLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        lock.writeLock().lock();
+        try {
+            return list.get(index);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        return list.size();
     }
```

Comment:
size is also a read operations which result is depending from the contents of the list, so we should lock it as well